contracts/yield-source/YearnV2YieldSource.sol

```diff
@@ -117,16 +117,16 @@ contract YearnV2YieldSource is IYieldSource, ERC20Upgradeable, OwnableUpgradeabl
     /// @dev Shares corresponding to the number of tokens supplied are mint to the user's balance
     /// @dev Asset tokens are supplied to the yield source, then deposited into Aave
     /// @param _amount The amount of asset tokens to be supplied
-    /// @param to The user whose balance will receive the tokens
+    /// @param to The user whose balance will receive the share tokens
     function supplyTokenTo(uint256 _amount, address to) external override nonReentrant {
-        uint256 shares = _tokenToShares(_amount);
-
-        _mint(to, shares);
-
-        // NOTE: we have to deposit after calculating shares to mint
+        
         token.safeTransferFrom(msg.sender, address(this), _amount);
 
-        _depositInVault();
+        uint256 amountDeposited = _depositInVault();
+        
+        uint256 shares = _tokenToShares(amountDeposited);
```

Comment:
Need to ensure this ordering is ok - removed comment says no but unit tests pass